sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -140,26 +140,36 @@
   public Span startSpan() {
     SpanContext parentContext = parent(parentType, parent, remoteParent);
     TraceId traceId;
+    SpanContext spanContext;
     SpanId spanId = idsGenerator.generateSpanId();
+    Decision samplingDecision;
     Tracestate tracestate = Tracestate.getDefault();
     if (parentContext == null || !parentContext.isValid()) {
       // New root span.
       traceId = idsGenerator.generateTraceId();
       // This is a root span so no remote or local parent.
       parentContext = null;
+      samplingDecision =
+          traceConfig.getSampler().shouldSample(parentContext, traceId, spanId, spanName, links);
+      spanContext =
+          SpanContext.create(
+              traceId,
+              spanId,
+              samplingDecision.isSampled() ? TRACE_OPTIONS_SAMPLED : TRACE_OPTIONS_NOT_SAMPLED,
+              tracestate);
     } else {
       // New child span.
       traceId = parentContext.getTraceId();
       tracestate = parentContext.getTracestate();
+      samplingDecision =
+          traceConfig.getSampler().shouldSample(parentContext, traceId, spanId, spanName, links);
+      spanContext =
+          SpanContext.createFromRemoteParent(
```

Comment:
how do we know this is remote? Could the parent be another internal span?

Reply:
Added [this](https://github.com/open-telemetry/opentelemetry-java/pull/656/files#diff-96d333cc34186523fe54031254bbceddR160) check here to see if parent type is remote parent then only use `createFromRemoteParent`.